dips/dip-178.md

```diff
@@ -0,0 +1,292 @@
+4. The merchant VASP submits a transaction on-chain and in parallel notify the customer VASP the refund is ready for settlement. Final validation will occur before this step (for example, verify the requested refund amount is not greater than the amount of original payment)
+
+---
+## Interface
+---
+This section describes a set of functionalities / activities that should be made available by the different VASPs in order to support P2M Refund flow based on this DIP. Each functionality can be conceptually compared to an API endpoint. Similar to API documentation, it includes the expected input and output for each functionality.
+
+## Command Types
+The following is a list of new and existing values for the `command_type` field of the `CommandRequestObject`. Each command type denotes a step in the refund payment off-chain negotiations and specifies the data of the requests and the response 
+* `InitRefundPayment` - Initializes a refund payment
+* `RefundReadyForSettlementNotification` - Notifies the payment is ready for settlement on-chain
+* `AbortRefund` - Aborts a payment
+
+| Command Type |Direction| Description | Request Data | Response Data |
+|-|-|-|-|-|
+|[InitRefundPayment](#initrefundpayment)|Merchant VASP > Customer VASP|This functionality allows the merchant VASP to init the refund. For example, this may occur after the merchant requested its VASP to refund the customer due to some items being out of stock|Reference ID ; Refund ID ; (optional) Amount|(optional) Recipient Signature|
+|[RefundReadyForSettlementNotification](#refundreadyforsettlementnotification)|Merchant VASP > Customer VASP|This functionality allows the Merchant VASP to indicate the refund is ready for settlement. The merchant VASP is expected to submit a transaction on-chain immediately before or after declaring it is ready|Reference ID ; Refund ID||
+|[AbortRefund](#abortrefund)|Any|This functionality allows both parties to abort the refund|Reference ID ; Refund ID||
+
+---
+## Refund Sequences
+---
+This section describes the expected sequences in P2M refund scenario.
+
+**Reminder:** Successful refund from the business to the customer requires a previous payment from the customer to the merchant. 
+
+### Successful Refund Sequence
+| Step | Command Type | Triggered by | Description |
+|-|-|-|-|
+|1|InitRefundPayment|Merchant VASP|Merchant VASP initializes the refund. Customer VASP will respond it is willing to accept the refund|
+|2|RefundReadyForSettlementNotification|Merchant VASP|The merchant VASP indicates the refund is ready for settlement. The VASP is expected to submit a transaction on-chain immediately before or after this step|
+
+#### Refund Sequence Diagram
+![Refund Sequence Diagram](https://static.swimlanes.io/b3f04660b9617526e07cd41cdb5ee8c1.png)
```

Comment:
I would place the `Refund Sequence Diagram` section higher up, above `Command Types`